### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by Kiransai-webonise**

### CODE REVIEW

**Comment by balakrishna-webonise** [comment]
It does not seem to be correct. CRUD should be there in the engine not here.

File: microblog/Gemfile [hunk 1]
```diff
@@ -6,6 +6,13 @@ ruby "3.0.0"
 # Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
 gem "rails", "~> 7.0.3", ">= 7.0.3.1"
 
+gem 'devise'
+
+# gem "blogtype", git: 'git@github.com:Kiransai-webonise/engine-rails.git',
+#                 branch: 'assignment3'
+
+gem "blogtype", path: '../../blogtype'
```
Comment (balakrishna-webonise) [inline microblog/Gemfile]: Do not commit this. This should be part of your local development.

**Review by balakrishna-webonise — CHANGES_REQUESTED** [review]

File: microblog/app/overrides/blogtypecontroller_override.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Blogtype::TypeBlogsController.class_eval do
+    after_action :allow_iframe, only: :lookup
```
Comment (balakrishna-webonise) [inline microblog/app/overrides/blogtypecontroller_override.rb]: why do you need this here?

**Review by balakrishna-webonise — CHANGES_REQUESTED** [review]

File: microblog/config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
+  devise_for :users
   resources :blogs
   put 'blogs/change_status/:id', to: 'blogs#change_status', as: 'change_status'
+
+  mount Articles::Engine, at: "/types"
```
Comment (balakrishna-webonise) [inline microblog/config/routes.rb]: the engine should be mounted on **static_pages’** right ?

File: microblog/Gemfile.lock [hunk 4]
```diff
@@ -1,3 +1,9 @@
+PATH
+  remote: ../../blogtype
```
Comment (balakrishna-webonise) [inline microblog/Gemfile.lock]: You should run bundle install once you update Gemfile. It's referring local path here.

**Review by balakrishna-webonise — CHANGES_REQUESTED** [review]

File: microblog/config/routes.rb (hunk 3 above)
Comment (Kiransai-webonise) [reply microblog/config/routes.rb]: `  mount Articles::Engine, at: "/static_pages"
`
Do you need like this?

Comment (balakrishna-webonise) [reply microblog/config/routes.rb]: Yes..

File: microblog/Gemfile.lock (hunk 4 above)
Comment (balakrishna-webonise) [reply microblog/Gemfile.lock]: > You should run bundle install once you update Gemfile. It's referring local path here.

Did you address this ?

**Review by balakrishna-webonise — APPROVED** [review]

**Pull request merged**